src/NServiceBus.Core/Serializers/MessageDeserializerResolver.cs

```diff
@@ -1,15 +1,24 @@
 ﻿namespace NServiceBus
 {
+    using System;
     using System.Collections.Generic;
-    using System.Linq;
     using Serialization;
 
     class MessageDeserializerResolver
     {
         public MessageDeserializerResolver(IMessageSerializer defaultSerializer, IEnumerable<IMessageSerializer> additionalDeserializers)
         {
             this.defaultSerializer = defaultSerializer;
-            serializersMap = additionalDeserializers.ToDictionary(key => key.ContentType, value => value);
+
+            foreach (var additionalDeserializer in additionalDeserializers)
+            {
+                if (serializersMap.ContainsKey(additionalDeserializer.ContentType))
+                {
+                    throw new Exception($"Multiple additional deserializers are registered for content-type '{additionalDeserializer.ContentType}'. Remove ambiguous deserializers for this content-type.");
```

Comment:
so you suggest to drop the "additional"? I kept that because it hints which API was causing this exception (since this will be a startup exception and you can actually have a default serializer using json while registering an additional json deserializer which will work).